crates/storage/store_db/in_memory.rs

```diff
@@ -281,156 +282,163 @@ impl StoreEngine for Store {
     }
 
     async fn add_account_code(&self, code_hash: H256, code: Bytes) -> Result<(), StoreError> {
-        self.inner().account_codes.insert(code_hash, code);
+        self.inner()?.account_codes.insert(code_hash, code);
         Ok(())
     }
 
     fn get_account_code(&self, code_hash: H256) -> Result<Option<Bytes>, StoreError> {
-        Ok(self.inner().account_codes.get(&code_hash).cloned())
+        Ok(self.inner()?.account_codes.get(&code_hash).cloned())
     }
 
     async fn set_chain_config(&self, chain_config: &ChainConfig) -> Result<(), StoreError> {
         // Store cancun timestamp
-        self.inner().chain_data.chain_config = Some(*chain_config);
+        self.inner()?.chain_data.chain_config = Some(*chain_config);
         Ok(())
     }
 
     fn get_chain_config(&self) -> Result<ChainConfig, StoreError> {
-        Ok(self.inner().chain_data.chain_config.unwrap())
+        match self.inner()?.chain_data.chain_config {
+            Some(chain_config) => Ok(chain_config),
+            None => Err(StoreError::Custom("No Chain Congif".to_string())),
+        }
```

Comment:
This is a common pattern to use `ok_or`.
```suggestion
        self.inner()?.chain_data.chain_config.ok_or(StoreError::Custom("No Chain Congif".to_string()))
```